File: src/utils/validate_choice.dart
import 'messages.dart';

void validateChoice(String choice, int min, int last, [String? message]) {
  if (choice.isEmpty || RegExp(r'[^0-9]').hasMatch(choice)) {
    throw Exception(messages['bad_choice']);
  }

  int parsedChoice = int.tryParse(choice) ?? 0;

  if (parsedChoice > last || parsedChoice < min) {
    throw Exception(message ?? messages['choice_not_found']);
  }
}

File: src/utils/validate-choice.js
import { messages } from "./messages.js";

export const validateChoice = (
  choice,
  min,
  last,
  message = messages.choice_not_found
) => {
  if (!choice || choice.length === 0 || /[^0-9]/.test(choice)) {
    throw new Error(messages.bad_choice);
  }
  if (+choice > last || +choice < min) {
    throw new Error(message);
  }
};
